pkg/resourcemanagers/topology.go

```diff
@@ -106,6 +123,17 @@ func (t *instanceTopologyManager) GetNodeTopology(ctx context.Context, instanceT
 	return nil, nil
 }
 
+// DoesInstanceTypeRequireResponse verifies whether or not we expect an instance to have an instance topology response.
+func (t *instanceTopologyManager) DoesInstanceTypeRequireResponse(instanceType string) bool {
```

Comment:
Should we include this check on L117 (i.e when `len(topologies == 0)`) to protect against scenarios where a deleting instance has its topology wiped? That way we don't incorrectly mark that type as unsupported for 24h

Reply:
I think that would be undesirable in the event that our regex is too greedy or there's a type added that matches and is not supported. That being said, I do think it probably requires some refactoring. Either:

1. Create a `instanceTopologyFound` set to track what instance types we've seen them in the past.
2. Don't add instance type to the unsupported list if it matches the regex.

Neither is ideal. It would be nice if we had the definitive response on whether the instance type is supported.

Reply:
Sounds good, thanks for making the change.